src/tests/ftest/util/network_utils.py

```diff
@@ -45,14 +45,14 @@ def __ne__(self, other):
 
     def __eq__(self, other):
         """Override the default implementation to compare devices."""
-        status = isinstance(other, NetworkDevice)
+        status = True
+        if not isinstance(other, NetworkDevice):
+            return False
         for key in self.__dict__:
-            if not status:
-                break
             try:
-                status &= getattr(self, key) == getattr(other, key)
+                status &= str(getattr(self, key)) == str(getattr(other, key))
             except AttributeError:
-                status = False
+                return False
```

Comment:
Optional - The original implementation had `if not status: break` and then `return status`. Which meant that if at any point `status` was `False`, the function would return `False`. So I think we can remove the `status` variable entirely and just directly return `False`